nilearn/_utils/niimg_conversions.py

```diff
@@ -142,13 +147,9 @@ def check_niimg(niimg, ensure_ndim=None, atleast_4d=False,
 
     Its application is idempotent.
     """
+
     # in case of an iterable
     if hasattr(niimg, "__iter__") and not isinstance(niimg, _basestring):
-        if ensure_ndim == 3:
-            raise TypeError(
-                "Data must be a 3D Niimg-like object but you provided a %s."
-                " See http://nilearn.github.io/building_blocks/"
-                "manipulating_mr_images.html#niimg." % type(niimg))
```

Comment:
I don't remove any feature here. I keep "going down" even if we are sure that the dimensionality is wrong in order to get a more accurate error message. The real TypeError (ie type of the data is wrong) is then handled in `load_niimg`, which is much more logical.